librosa/core/constantq.py

```diff
@@ -612,115 +604,80 @@ def icqt(C, sr=22050, hop_length=512, fmin=None,
     >>> y_hat = librosa.icqt(C=C, sr=sr, hop_length=hop_length,
     ...                 bins_per_octave=bins_per_octave)
     '''
-    warnings.warn('librosa.icqt is unstable, and subject to change in future versions. '
-                  'Please use with caution.')
-
-    n_bins, n_frames = C.shape
-    n_octaves = int(np.ceil(float(n_bins) / bins_per_octave))
-
-    if amin is None:
-        amin = util.tiny(C)
 
     if fmin is None:
         fmin = note_to_hz('C1')
 
-    freqs = cqt_frequencies(n_bins,
-                            fmin,
+    # Get the top octave of frequencies
+    n_bins = len(C)
+
+    freqs = cqt_frequencies(n_bins, fmin,
                             bins_per_octave=bins_per_octave,
                             tuning=tuning)[-bins_per_octave:]
 
-    fmin_t = np.min(freqs)
-
-    # Make the filter bank
-    basis, lengths = filters.constant_q(sr=sr,
-                                        fmin=fmin_t,
-                                        n_bins=bins_per_octave,
-                                        bins_per_octave=bins_per_octave,
-                                        filter_scale=filter_scale,
-                                        tuning=tuning,
-                                        norm=norm,
-                                        window=window,
-                                        pad_fft=True)
-    n_fft = basis.shape[1]
+    fmin_t = np.min(freqs[-bins_per_octave:])
```

Comment:
not sure why you need to slice the array here? isn't `freqs` already `bins_per_octave` long from the line above?